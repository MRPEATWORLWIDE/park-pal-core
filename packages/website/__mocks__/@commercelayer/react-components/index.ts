export * from '@commercelayer/react-components/lib/index'