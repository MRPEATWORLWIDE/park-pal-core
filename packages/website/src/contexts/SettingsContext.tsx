import { getRawDataOrganization } from '#data/organization'
import { getLocale, Locale } from '#i18n/locale'
import type { RawDataOrganization } from '@commercelayer/demo-store-types'
import memoize from 'lodash/memoize'
import { createContext, useContext } from 'react'

type Context = {
  organization: RawDataOrganization
  locale: Locale
}

const SettingsContext = createContext<Partial<Context>>({})

export const SettingsProvider: React.FC<Context> = ({ children, ...props }) => (
  <SettingsContext.Provider value={{ ...props }}>
    {children}
  </SettingsContext.Provider>
)

export const useSettingsContext = () => useContext(SettingsContext)

export const serverSideSettings = memoize(
  async (localeCode: string): Promise<SettingsContextProps> => {
    return {
      settingsContext: {
        organization: await getRawDataOrganization(),
        locale: await getLocale(localeCode),
      }
    }
  }
)

type SettingsContextProps = {
  settingsContext: Context
}
