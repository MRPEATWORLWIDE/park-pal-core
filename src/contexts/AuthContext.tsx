import { createContext, useContext } from 'react'

type Context = {
  accessToken: string
  domain: string
  endpoint: string
  organization: string
}

const AuthContext = createContext<Partial<Context>>({})

export const AuthProvider: React.FC<Context> = ({ children, ...props }) => (
  <AuthContext.Provider value={{ ...props }}>
    {children}
  </AuthContext.Provider>
)

export const useAuthContext = () => useContext(AuthContext)
