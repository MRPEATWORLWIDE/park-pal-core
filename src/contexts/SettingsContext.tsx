import { getRawDataOrganization, RawDataOrganization } from '#data/organization'
import { createContext, useContext } from 'react'

type Context = {
  organization: RawDataOrganization
}

const SettingsContext = createContext<Partial<Context>>({})

export const SettingsProvider: React.FC<Context> = ({ children, ...props }) => (
  <SettingsContext.Provider value={{ ...props }}>
    {children}
  </SettingsContext.Provider>
)

export const useSettingsContext = () => useContext(SettingsContext)

export const serverSideSettings = async (): Promise<SettingsContextProps> => {
  return {
    settingsContext: {
      organization: await getRawDataOrganization()
    }
  }
}

type SettingsContextProps = {
  settingsContext: Context
}
