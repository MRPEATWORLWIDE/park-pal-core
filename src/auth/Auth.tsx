import { getLocale } from '#i18n/locale'
import { AuthReturnType, ClientCredentials, getSalesChannelToken } from '@commercelayer/js-auth'
import { CommerceLayer } from '@commercelayer/react-components'
import { useRouter } from 'next/router'
import { FC, useEffect, useMemo, useState } from 'react'

type Auth = {
  accessToken: string
  refreshToken?: string
  expires?: number
  tokenType: string
}

const getClientCredentials = (market: number): ClientCredentials => ({
  clientId: process.env.NEXT_PUBLIC_CL_CLIENT_ID,
  endpoint: process.env.NEXT_PUBLIC_CL_ENDPOINT,
  scope: `market:${market}`
})

const getAuth = (market: number): Auth | null => {
  const cookieKey = getCookieKey(market)
  return JSON.parse(localStorage.getItem(cookieKey) || 'null')
}

const storeAuth = (market: number, authReturn: Awaited<AuthReturnType>): Auth | null => {
  if (!authReturn) {
    return null
  }

  const cookieKey = getCookieKey(market)

  const auth: Auth = {
    accessToken: authReturn.accessToken,
    refreshToken: authReturn.refreshToken,
    expires: authReturn.expires?.getTime(),
    tokenType: authReturn.tokenType
  }

  localStorage.setItem(cookieKey, JSON.stringify(auth))

  return auth
}

const getCookieKey = <M extends number>(market: M): `clayer_access_token-market:${M}` => `clayer_access_token-market:${market}`

const hasExpired = (time: number | undefined): boolean => time === undefined || time < Date.now()

const isValid = (auth: Auth | null): auth is Auth => !hasExpired(auth?.expires)

export const Auth: FC = ({ children }) => {

  const router = useRouter()

  const locale = useMemo(() => getLocale(router.query.locale as string), [router])

  const [market, setMarket] = useState<number | undefined>(locale?.country?.market)
  const [auth, setAuth] = useState<Auth | null>(null)

  useEffect(function updateMarket() {
    if (locale?.country?.market !== market) {
      setMarket(locale?.country?.market)
    }
  }, [locale])

  useEffect(function updateAccessToken() {
    if (market === undefined) {
      setAuth(null)
      return
    }

    const storedAuth: Auth | null = getAuth(market)
    const authIsValid = isValid(storedAuth)

    if (authIsValid) {
      setAuth(storedAuth)
    } else {
      getSalesChannelToken( getClientCredentials(market) )
        .then( authReturn => setAuth( storeAuth(market, authReturn) ) )
    }
  }, [market])

  if (!auth) {
    return <>{children}</>
  }

  return (
    <CommerceLayer accessToken={auth.accessToken} endpoint={process.env.NEXT_PUBLIC_CL_ENDPOINT}>
      {children}
    </CommerceLayer>
  )
}
