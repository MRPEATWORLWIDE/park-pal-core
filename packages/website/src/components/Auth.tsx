import { AuthProvider } from '#contexts/AuthContext'
import { useSettingsContext } from '#contexts/SettingsContext'
import { AuthReturnType, ClientCredentials, getSalesChannelToken } from '@commercelayer/js-auth'
import { CommerceLayer } from '@commercelayer/react-components'
import { useRouter } from 'next/router'
import { useEffect, useState } from 'react'

type Auth = {
  accessToken: string
  refreshToken?: string
  expires?: number
  tokenType: string
}

const getClientCredentials = (clientId: string, endpoint: string, market: number): ClientCredentials => ({
  clientId,
  endpoint,
  scope: `market:${market}`
})

const getAuth = (market: number): Auth | null => {
  const storeKey = getStoreKey(market)
  return JSON.parse(localStorage.getItem(storeKey) || 'null')
}

const storeAuth = (market: number, authReturn: Awaited<AuthReturnType>): Auth | null => {
  if (!authReturn) {
    return null
  }

  const storeKey = getStoreKey(market)

  const auth: Auth = {
    tokenType: authReturn.tokenType,
    accessToken: authReturn.accessToken,
    expires: authReturn.expires?.getTime(),
    refreshToken: authReturn.refreshToken
  }

  localStorage.setItem(storeKey, JSON.stringify(auth))

  return auth
}

const getStoreKey = <M extends number>(market: M): `clayer_token-market:${M}` => `clayer_token-market:${market}`

const hasExpired = (time: number | undefined): boolean => time === undefined || time < Date.now()

const isValid = (auth: Auth | null): auth is Auth => !hasExpired(auth?.expires)

export const Auth: React.FC<{}> = ({ children }) => {

  const router = useRouter()
  const settings = useSettingsContext()

  const [market, setMarket] = useState<number | undefined>(settings.locale?.isShoppable ? settings.locale?.country.market : undefined)
  const [auth, setAuth] = useState<Auth | null>(null)

  const clientId = process.env.NEXT_PUBLIC_CL_CLIENT_ID
  const endpoint = process.env.NEXT_PUBLIC_CL_ENDPOINT

  useEffect(function updateMarket() {
    if (settings.locale?.isShoppable && settings.locale?.country.market !== market) {
      setMarket(settings.locale?.country.market)
    }
  }, [settings.locale, market])

  useEffect(function updateAccessToken() {
    let isMounted = true

    if (market === undefined || clientId === undefined || endpoint === undefined) {
      setAuth(null)
      return
    }

    const storedAuth: Auth | null = getAuth(market)
    const authIsValid = isValid(storedAuth)

    if (authIsValid) {
      setAuth(storedAuth)
    } else {
      getSalesChannelToken(getClientCredentials(clientId, endpoint, market))
        .then(authReturn => {
          if (isMounted) {
            setAuth(storeAuth(market, authReturn))
          }
        })
    }

    return () => {
      isMounted = false
    }
  }, [market, router.asPath, clientId, endpoint])

  if (!auth || !endpoint) {
    return <>{children}</>
  }

  const { hostname } = new URL(endpoint)
  const [, organization, domain] = hostname.match(/^(.*).(commercelayer.(co|io))$/) || []

  return (
    <AuthProvider accessToken={auth.accessToken} endpoint={endpoint} organization={organization} domain={domain}>
      <CommerceLayer accessToken={auth.accessToken} endpoint={endpoint}>
        <>{children}</>
      </CommerceLayer>
    </AuthProvider>
  )
}
