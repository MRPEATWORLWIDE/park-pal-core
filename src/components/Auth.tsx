import { defaultLocale, getLocale } from '#i18n/locale'
import { AuthReturnType, ClientCredentials, getSalesChannelToken } from '@commercelayer/js-auth'
import { CommerceLayer } from '@commercelayer/react-components'
import { useRouter } from 'next/router'
import { useEffect, useMemo, useState } from 'react'
import { AuthProvider } from '#contexts/AuthContext'

type Auth = {
  accessToken: string
  refreshToken?: string
  expires?: number
  tokenType: string
}

const getClientCredentials = (market: number): ClientCredentials => ({
  clientId: process.env.NEXT_PUBLIC_CL_CLIENT_ID,
  endpoint: process.env.NEXT_PUBLIC_CL_ENDPOINT,
  scope: `market:${market}`
})

const getAuth = (market: number): Auth | null => {
  const storeKey = getStoreKey(market)
  return JSON.parse(localStorage.getItem(storeKey) || 'null')
}

const storeAuth = (market: number, authReturn: Awaited<AuthReturnType>): Auth | null => {
  if (!authReturn) {
    return null
  }

  const storeKey = getStoreKey(market)

  const auth: Auth = {
    tokenType: authReturn.tokenType,
    accessToken: authReturn.accessToken,
    expires: authReturn.expires?.getTime(),
    refreshToken: authReturn.refreshToken
  }

  localStorage.setItem(storeKey, JSON.stringify(auth))

  return auth
}

const getStoreKey = <M extends number>(market: M): `clayer_token-market:${M}` => `clayer_token-market:${market}`

const hasExpired = (time: number | undefined): boolean => time === undefined || time < Date.now()

const isValid = (auth: Auth | null): auth is Auth => !hasExpired(auth?.expires)

export const Auth: React.FC<{}> = ({ children }) => {

  const router = useRouter()

  const locale = useMemo(() => getLocale(router.query.locale || defaultLocale), [router])

  const [market, setMarket] = useState<number | undefined>(locale.isShoppable ? locale.country.market : undefined)
  const [auth, setAuth] = useState<Auth | null>(null)

  const endpoint = process.env.NEXT_PUBLIC_CL_ENDPOINT
  const { hostname } = new URL(endpoint)
  const [, organization, domain] = hostname.match(/^(.*).(commercelayer.(co|io))$/) || []

  useEffect(function updateMarket() {
    if (locale.isShoppable && locale.country.market !== market) {
      setMarket(locale.country.market)
    }
  }, [locale, market])

  useEffect(function updateAccessToken() {
    let isMounted = true

    if (market === undefined) {
      setAuth(null)
      return
    }

    const storedAuth: Auth | null = getAuth(market)
    const authIsValid = isValid(storedAuth)

    if (authIsValid) {
      setAuth(storedAuth)
    } else {
      getSalesChannelToken(getClientCredentials(market))
        .then(authReturn => {
          if (isMounted) {
            setAuth(storeAuth(market, authReturn))
          }
        })
    }

    return () => {
      isMounted = false
    }
  }, [market, router.asPath])

  if (!auth) {
    return <>{children}</>
  }

  return (
    <AuthProvider accessToken={auth.accessToken} endpoint={endpoint} organization={organization} domain={domain}>
      <CommerceLayer accessToken={auth.accessToken} endpoint={endpoint}>
        <>{children}</>
      </CommerceLayer>
    </AuthProvider>
  )
}
