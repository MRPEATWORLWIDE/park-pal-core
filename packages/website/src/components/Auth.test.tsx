import { useAuthContext } from '#contexts/AuthContext'
import { SettingsProvider } from '#contexts/SettingsContext'
import CommerceLayerContext from '../../../../node_modules/@commercelayer/react-components/lib/cjs/context/CommerceLayerContext'
import { act, render, screen } from '@testing-library/react'
import { createLocale, createOrganization, createRouter } from 'jest.helpers'
import { useContext } from 'react'
import { Auth } from './Auth'

import * as JsAuth from '@commercelayer/js-auth'

jest.mock('@commercelayer/js-auth', () => {
  return {
    __esModule: true,
    ...jest.requireActual('@commercelayer/js-auth')
  };
});

const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const authentication = jest.spyOn(JsAuth, 'authenticate')

beforeEach(() => {
  useRouter.mockReset()
})

const ContextTester = () => {
  const { accessToken: clAccessToken, endpoint: clEndpoint } = useContext(CommerceLayerContext)
  const { accessToken, endpoint, organization, domain } = useAuthContext()

  return (
    <>
      <div data-testid="clAccessToken">clAccessToken: {clAccessToken}</div>
      <div data-testid="clEndpoint">clEndpoint: {clEndpoint}</div>
      <div data-testid="accessToken">accessToken: {accessToken}</div>
      <div data-testid="endpoint">endpoint: {endpoint}</div>
      <div data-testid="organization">organization: {organization}</div>
      <div data-testid="domain">domain: {domain}</div>
    </>
  )
}

test('should match the snapshot', async () => {
  process.env.NEXT_PUBLIC_CL_CLIENT_ID = 'client-1234'

  useRouter.mockImplementation(() => createRouter('/'))

  authentication.mockResolvedValue({
    tokenType: 'bearer',
    accessToken: accessToken,
    expires: new Date(0),
    refreshToken: 'refreshToken-1234',
    createdAt: 0,
    expiresIn: 0,
    ownerId: '',
    ownerType: 'customer',
    scope: '',
  })

  let container

  await act(async () => {
    ({ container } = render(
      <SettingsProvider locale={createLocale()} organization={{ ...createOrganization() }}>
        <Auth locale={createLocale()}><ContextTester /></Auth>
      </SettingsProvider>
    ))
  })

  expect(container).toMatchSnapshot()
})

test('should fetch accessToken and set it properly when "locale" is set', async () => {
  process.env.NEXT_PUBLIC_CL_CLIENT_ID = 'client-1234'

  useRouter.mockImplementation(() => createRouter('/'))

  authentication.mockResolvedValue({
    tokenType: 'bearer',
    accessToken: accessToken,
    expires: new Date(0),
    refreshToken: 'refreshToken-1234',
    createdAt: 0,
    expiresIn: 0,
    ownerId: '',
    ownerType: 'customer',
    scope: '',
  })

  await act(async () => {
    render(
      <SettingsProvider locale={createLocale()} organization={{ ...createOrganization() }}>
        <Auth locale={createLocale()}><ContextTester /></Auth>
      </SettingsProvider>
    )
  })

  expect(authentication).toHaveBeenCalledWith('client_credentials', {
    clientId: 'client-1234',
    scope: 'market:code:us'
  })

  expect(localStorage.getItem('clayer_token-market:code:us')).toEqual(JSON.stringify({
    tokenType: 'bearer',
    accessToken: accessToken,
    expires: 0,
    refreshToken: 'refreshToken-1234'
  }))

  expect(screen.getByTestId('clAccessToken')).toHaveTextContent(accessToken)
  expect(screen.getByTestId('clEndpoint')).toHaveTextContent('https://demo-store.commercelayer.io')

  expect(screen.getByTestId('accessToken')).toHaveTextContent(accessToken)
  expect(screen.getByTestId('endpoint')).toHaveTextContent('https://demo-store.commercelayer.io')
  expect(screen.getByTestId('organization')).toHaveTextContent('demo-store')
  expect(screen.getByTestId('domain')).toHaveTextContent('commercelayer.io')

})

const accessToken = 'eyJhbGciOiJSUzUxMiIsInR5cCI6IkpXVCIsImtpZCI6IjliN2JiZmVlMzQzZDVkNDQ5ZGFkODhmMjg0MGEyZTM3YzhkZWFlZTg5NjM4MGQ1ODA2YTc4NWVkMWQ1OTc5ZjAifQ.eyJvcmdhbml6YXRpb24iOnsiaWQiOiJleW9aT0Z2UHBSIiwic2x1ZyI6ImRlbW8tc3RvcmUiLCJlbnRlcnByaXNlIjp0cnVlLCJyZWdpb24iOiJldS13ZXN0LTEifSwiYXBwbGljYXRpb24iOnsiaWQiOiJwWWRxaVBBUW5NIiwiY2xpZW50X2lkIjoiQklTRzhiYjNHV3BDOF9EN050MVN1V1dkaWVTNWJKcTgzMUE1MExnQl9JZyIsImtpbmQiOiJzYWxlc19jaGFubmVsIiwicHVibGljIjp0cnVlfSwic2NvcGUiOiJzdG9ja19sb2NhdGlvbjppZDpER3pBb3VwcHduIiwiZXhwIjoxNzI3MTY5NTQ5LCJ0ZXN0Ijp0cnVlLCJyYW5kIjowLjg0MjUwNTQ0NTgzNjAxOTksImlhdCI6MTcyNzE2MjM0OSwiaXNzIjoiaHR0cHM6Ly9hdXRoLmNvbW1lcmNlbGF5ZXIuaW8ifQ.rFYL56S-j60pgJ5QaomqHZF3fk97jRAOpj2kzHMr6TbYdwyJUc1GFxQZuHkN0CLuVh32hQ_nsJ6alw7zJiCcmlkqKhlTHqvyHHcYvmFkGP9ZhiBOvfVMNSjHkxcWT-a58U6ao8UZflqSkYkOAqWTQK3rQuAQTWtSox18UstUnI_G8oxtzRLmKDJhCzC5Bl3ewwNgrEo7soosRq9XTSjOffS57JLCl3-3wy69fhHAmmk_zXaUuV_vgiLREh-WYE4fqBg7dUnarqpE7w_20bHclEQ_Zg9dAaZ093g0IeHOs-GtzZyCDhSbcgJgp62wPQomjiJxKH1yjBJ5EkttqqSzQdUJftL8MgfEUF8tfPBMTs6GQSxIczPNmlk4EMeeIpKA8TMj0pPzwXOtd8Eetqen9DzgKYWoKLUXRSZx41YcAdHOFMSEh4j6lGL3GBVP2KiuyLzwk_sHKQR_4ppGDY5aIQA7pYjSyQVDf2SQa07FhSsZNfWdIkrv-YpjSDeNA0zDuoJkIVG4yUuE8D-LHYM4_P9KJPKo9kmd00zJthkcAZ-_QoBxq5-QPLh4aWiCwhRzfeJVtsIdFfBeMos4ar36g-5XtYjJ1c259TsHPF5iu38n4fbBY1jYFle1eiANAWpmPw_GVi13HutEe7Aoef0zFGzK3mEulls2NcBu0Ybh3as'
