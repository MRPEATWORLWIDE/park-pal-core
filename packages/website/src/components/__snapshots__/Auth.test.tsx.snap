// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <div
    data-testid="clAccessToken"
  >
    clAccessToken: 
    eyJhbGciOiJSUzUxMiIsInR5cCI6IkpXVCIsImtpZCI6IjliN2JiZmVlMzQzZDVkNDQ5ZGFkODhmMjg0MGEyZTM3YzhkZWFlZTg5NjM4MGQ1ODA2YTc4NWVkMWQ1OTc5ZjAifQ.eyJvcmdhbml6YXRpb24iOnsiaWQiOiJleW9aT0Z2UHBSIiwic2x1ZyI6ImRlbW8tc3RvcmUiLCJlbnRlcnByaXNlIjp0cnVlLCJyZWdpb24iOiJldS13ZXN0LTEifSwiYXBwbGljYXRpb24iOnsiaWQiOiJwWWRxaVBBUW5NIiwiY2xpZW50X2lkIjoiQklTRzhiYjNHV3BDOF9EN050MVN1V1dkaWVTNWJKcTgzMUE1MExnQl9JZyIsImtpbmQiOiJzYWxlc19jaGFubmVsIiwicHVibGljIjp0cnVlfSwic2NvcGUiOiJzdG9ja19sb2NhdGlvbjppZDpER3pBb3VwcHduIiwiZXhwIjoxNzI3MTY5NTQ5LCJ0ZXN0Ijp0cnVlLCJyYW5kIjowLjg0MjUwNTQ0NTgzNjAxOTksImlhdCI6MTcyNzE2MjM0OSwiaXNzIjoiaHR0cHM6Ly9hdXRoLmNvbW1lcmNlbGF5ZXIuaW8ifQ.rFYL56S-j60pgJ5QaomqHZF3fk97jRAOpj2kzHMr6TbYdwyJUc1GFxQZuHkN0CLuVh32hQ_nsJ6alw7zJiCcmlkqKhlTHqvyHHcYvmFkGP9ZhiBOvfVMNSjHkxcWT-a58U6ao8UZflqSkYkOAqWTQK3rQuAQTWtSox18UstUnI_G8oxtzRLmKDJhCzC5Bl3ewwNgrEo7soosRq9XTSjOffS57JLCl3-3wy69fhHAmmk_zXaUuV_vgiLREh-WYE4fqBg7dUnarqpE7w_20bHclEQ_Zg9dAaZ093g0IeHOs-GtzZyCDhSbcgJgp62wPQomjiJxKH1yjBJ5EkttqqSzQdUJftL8MgfEUF8tfPBMTs6GQSxIczPNmlk4EMeeIpKA8TMj0pPzwXOtd8Eetqen9DzgKYWoKLUXRSZx41YcAdHOFMSEh4j6lGL3GBVP2KiuyLzwk_sHKQR_4ppGDY5aIQA7pYjSyQVDf2SQa07FhSsZNfWdIkrv-YpjSDeNA0zDuoJkIVG4yUuE8D-LHYM4_P9KJPKo9kmd00zJthkcAZ-_QoBxq5-QPLh4aWiCwhRzfeJVtsIdFfBeMos4ar36g-5XtYjJ1c259TsHPF5iu38n4fbBY1jYFle1eiANAWpmPw_GVi13HutEe7Aoef0zFGzK3mEulls2NcBu0Ybh3as
  </div>
  <div
    data-testid="clEndpoint"
  >
    clEndpoint: 
    https://demo-store.commercelayer.io
  </div>
  <div
    data-testid="accessToken"
  >
    accessToken: 
    eyJhbGciOiJSUzUxMiIsInR5cCI6IkpXVCIsImtpZCI6IjliN2JiZmVlMzQzZDVkNDQ5ZGFkODhmMjg0MGEyZTM3YzhkZWFlZTg5NjM4MGQ1ODA2YTc4NWVkMWQ1OTc5ZjAifQ.eyJvcmdhbml6YXRpb24iOnsiaWQiOiJleW9aT0Z2UHBSIiwic2x1ZyI6ImRlbW8tc3RvcmUiLCJlbnRlcnByaXNlIjp0cnVlLCJyZWdpb24iOiJldS13ZXN0LTEifSwiYXBwbGljYXRpb24iOnsiaWQiOiJwWWRxaVBBUW5NIiwiY2xpZW50X2lkIjoiQklTRzhiYjNHV3BDOF9EN050MVN1V1dkaWVTNWJKcTgzMUE1MExnQl9JZyIsImtpbmQiOiJzYWxlc19jaGFubmVsIiwicHVibGljIjp0cnVlfSwic2NvcGUiOiJzdG9ja19sb2NhdGlvbjppZDpER3pBb3VwcHduIiwiZXhwIjoxNzI3MTY5NTQ5LCJ0ZXN0Ijp0cnVlLCJyYW5kIjowLjg0MjUwNTQ0NTgzNjAxOTksImlhdCI6MTcyNzE2MjM0OSwiaXNzIjoiaHR0cHM6Ly9hdXRoLmNvbW1lcmNlbGF5ZXIuaW8ifQ.rFYL56S-j60pgJ5QaomqHZF3fk97jRAOpj2kzHMr6TbYdwyJUc1GFxQZuHkN0CLuVh32hQ_nsJ6alw7zJiCcmlkqKhlTHqvyHHcYvmFkGP9ZhiBOvfVMNSjHkxcWT-a58U6ao8UZflqSkYkOAqWTQK3rQuAQTWtSox18UstUnI_G8oxtzRLmKDJhCzC5Bl3ewwNgrEo7soosRq9XTSjOffS57JLCl3-3wy69fhHAmmk_zXaUuV_vgiLREh-WYE4fqBg7dUnarqpE7w_20bHclEQ_Zg9dAaZ093g0IeHOs-GtzZyCDhSbcgJgp62wPQomjiJxKH1yjBJ5EkttqqSzQdUJftL8MgfEUF8tfPBMTs6GQSxIczPNmlk4EMeeIpKA8TMj0pPzwXOtd8Eetqen9DzgKYWoKLUXRSZx41YcAdHOFMSEh4j6lGL3GBVP2KiuyLzwk_sHKQR_4ppGDY5aIQA7pYjSyQVDf2SQa07FhSsZNfWdIkrv-YpjSDeNA0zDuoJkIVG4yUuE8D-LHYM4_P9KJPKo9kmd00zJthkcAZ-_QoBxq5-QPLh4aWiCwhRzfeJVtsIdFfBeMos4ar36g-5XtYjJ1c259TsHPF5iu38n4fbBY1jYFle1eiANAWpmPw_GVi13HutEe7Aoef0zFGzK3mEulls2NcBu0Ybh3as
  </div>
  <div
    data-testid="endpoint"
  >
    endpoint: 
    https://demo-store.commercelayer.io
  </div>
  <div
    data-testid="organization"
  >
    organization: 
    demo-store
  </div>
  <div
    data-testid="domain"
  >
    domain: 
    commercelayer.io
  </div>
</div>
`;
